tasks/simplification.py

```diff
@@ -0,0 +1,227 @@
+                                          shp_filename(self.schema, self.table, SIMPLIFIED_SUFFIX)))
+        shell(cmd)
+        self.executed = True
+
+    def output(self):
+        return PostgresTarget(self.schema, self.table)
+
+    def complete(self):
+        if not self.executed:
+            return False
+        return self.output().exists()
+
+
+class SimplifyGeometriesMapshaper(WrapperTask):
+    schema = Parameter()
+    table = Parameter()
+    retainpercentage = Parameter(default=DEFAULT_M_RETAIN_PERCENTAGE)
+    skipfailures = Parameter(default=SKIPFAILURES_NO)
+
+    def requires(self):
+        yield ImportSimplifiedShapefile(schema=self.schema, table=self.table, retainpercentage=self.retainpercentage,
+                                        skipfailures=self.skipfailures)
+
+
+def postgis_simplification_factor(schema, table, geomfield, divisor_power):
+    session = CurrentSession().get()
+    return session.execute('SELECT AVG(ST_Perimeter({geomfield}) / ST_NPoints({geomfield})) / 10 ^ ({divisor} / 10) '
+                           'from "{schema}".{table}'.format(
+                            schema=schema, table=table, geomfield=geomfield, divisor=divisor_power
+                           )).fetchone()[0]
+
+
+class SimplifyGeometriesPostGIS(Task):
+    schema = Parameter()
+    table = Parameter()
+    geomfield = Parameter(default=DEFAULT_GEOMFIELD)
+    retainfactor = Parameter(default=UNKNOWN_RETAIN_FACTOR)
+
+    executed = False
+
+    def run(self):
+        session = CurrentSession().get()
+
+        if self.retainfactor == '?':
+            self.retainfactor = DEFAULT_P_RETAIN_FACTOR
+
+        factor = postgis_simplification_factor(self.schema, self.table, self.geomfield, self.retainfactor)
+
+        session.execute('UPDATE "{schema}".{table} '
+                        'SET {geomfield} = ST_MakeValid(ST_SimplifyVW({geomfield}, {factor})); '.format(
+                            schema=self.schema, table=self.table, geomfield=self.geomfield, factor=factor))
+        session.execute("COMMIT")
+
+        self.executed = True
+
+    def output(self):
+        return PostgresTarget(self.schema, self.table)
+
+    def complete(self):
+        if not self.executed:
+            return False
+        return self.output().exists()
+
+
+class MapshaperSimplification():
+    def __init__(self, retainpercentage=DEFAULT_M_RETAIN_PERCENTAGE, skipfailures=SKIPFAILURES_NO):
+        self.retainpercentage = retainpercentage
+        self.skipfailures = skipfailures
+
+
+class PostGISSimplification():
+    def __init__(self, geomfield=DEFAULT_GEOMFIELD, retainfactor=UNKNOWN_RETAIN_FACTOR):
+        self.geomfield = geomfield
+        self.retainfactor = retainfactor
+
+
+class Simplify(Task):
+    schema = Parameter()
+
+    dont_simplify = None
+    '''
+    Array of names of tables in the schema that we don't want to simplify
+    Example: dont_simplify = ['mytable1', 'mytable2']
+    '''
```

Comment:
Thinking on how to integrate this, I'm not sure if a table that simplifies everything that has not been skipped is a good idea. For example, in the US census, you would have to skip all the intermediate tables of the clipping part.

Also, most tasks typically work with a single table, and this one works with the entire schema. I think it can be tricky to create a `TableTask` that takes the input from this simplification task as an input.

Maybe I'm missing context. Can you provide an example on how to use this for one dataset?